include/flash.h

```diff
@@ -127,6 +165,47 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+
+struct flash_pages_info {
+	off_t start_offset;
+	size_t size;
+};
+
+/**
+ *  @brief  Get size and start offset of flash page at certain flash offset.
```

Comment:
@andrzej-kaczmarek What is your opinion - should offset or absolute address being returned?